packages/design-system/src/components/Modal/Modal.tsx

```diff
@@ -33,12 +33,13 @@ export type ModalPropsType = {
 		description?: string;
 		icon?: IconProp;
 	};
-	onClose?: Function;
+	onClose?: () => void;
 	disclosure?: ReactElement;
 	primaryAction?: PrimaryActionPropsType;
 	secondaryAction?: ButtonSecondaryPropsType;
 	preventEscaping?: boolean;
 	children: ReactNode | ReactNode[];
+	dataFeature?: string;
 };
```

Comment:
```suggestion
} & Omit<HTMLAttributes<HTMLDivElement>, 'className' | 'style'>;
```
We should be able to spread whatever data-thing we want on there :)